integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java

```diff
@@ -134,23 +127,17 @@
    */
   public CarbonTableCacheModel getCarbonCache(SchemaTableName table, String location,
       Configuration config) {
-    if (!carbonCache.get().containsKey(table) || carbonCache.get().get(table) == null) {
-      updateSchemaTables(table, config);
-      parseCarbonMetadata(table, location, config);
-    }
-    if (carbonCache.get().containsKey(table)) {
-      return carbonCache.get().get(table);
-    } else {
-      return null;
+    updateSchemaTables(table, config);
+    CarbonTableCacheModel carbonTableCacheModel = carbonCache.get().get(table);
+    if (carbonTableCacheModel == null || !carbonTableCacheModel.isValid()) {
+      return parseCarbonMetadata(table, location, config);
     }
+    return carbonTableCacheModel;
   }
 
   private void removeTableFromCache(SchemaTableName table) {
```

Comment:
removeTableFromCache, is not removing any table from cache. Need to remove here or rename the method ?

Reply:
ok, removed the method